core/services/head_tracker.go

```diff
@@ -267,15 +269,21 @@ func (ht *HeadTracker) listenForNewHeads() {
 		logger.WarnIf(errors.Wrap(err, "HeadTracker failed when unsubscribe from head"))
 	}()
 
+	ctx, cancel := utils.ContextFromChan(ht.done)
+	defer cancel()
+
 	for {
 		if !ht.subscribe() {
-			return
+			break
```

Comment:
Why change this, doesn't it do the same thing?

Reply:
personal preferences, feels more local